ntp-udp/src/lib.rs

```diff
@@ -47,18 +49,34 @@ impl UdpSocket {
         peer_addr = debug(self.as_ref().peer_addr()),
         buf_size = buf.len(),
     ))]
-    pub async fn send(&self, buf: &[u8]) -> io::Result<usize> {
+    pub async fn send(&self, buf: &[u8]) -> io::Result<(usize, Option<NtpTimestamp>)> {
+        let send_size = self.send_help(buf).await?;
+
+        // the send timestamp may never come (when the driver does not support send timestamping)
+        // set a very short timeout to prevent hanging forever. We automatically fall back to a
+        // less accurate timestamp when this function returns None
+        let timeout = std::time::Duration::from_millis(10);
```

Comment:
This timeout turns out to be a bit too low on my machine, maybe we should just make this 100 milliseconds.

Reply:
sounds good

Reply:
we'll need to look at this in more detail. bumping that timeout makes various tests fail

Reply:
This seems to have been a symptom of waiting for the wrong queue (both of these actually), so don't think this needs further attention for now.